hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -222,53 +230,188 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
 
   @Override
   public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
-    if (!isValidIdentifier(from)) {
-      throw new NoSuchTableException("Invalid identifier: %s", from);
+    renameEntity(from, originalTo, true);
+  }
+
+  @Override
+  public boolean dropView(TableIdentifier identifier) {
```

Comment:
From https://github.com/apache/iceberg/issues/9433, it looks like we need to delete the metadata file?